File: 2023-2024/sourcecode/mod3/ResourcePatternBasicsJava/app/src/main/java/com/example/resourcepatternbasicsjava/ui/CommentComparator.java
package com.example.resourcepatternbasicsjava.ui;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.example.resourcepatternbasicsjava.data.CommentResponseItem;

public class CommentComparator extends DiffUtil.ItemCallback<CommentResponseItem> {
    @Override
    public boolean areItemsTheSame(@NonNull CommentResponseItem oldItem, @NonNull CommentResponseItem newItem) {
        return oldItem == newItem;
    }

    @Override
    public boolean areContentsTheSame(CommentResponseItem oldItem, @NonNull CommentResponseItem newItem) {
        return oldItem.equals(newItem);
    }
}

File: 2023-2024/sourcecode/mod3/ResourcePatternBasicsKotlin/app/src/main/java/com/example/resourcepatternbasicskotlin/ui/CommentComparator.kt
package com.example.resourcepatternbasicskotlin.ui

import androidx.recyclerview.widget.DiffUtil
import com.example.resourcepatternbasicskotlin.data.CommentResponseItem

class CommentComparator : DiffUtil.ItemCallback<CommentResponseItem>() {
    override fun areItemsTheSame(oldItem: CommentResponseItem, newItem: CommentResponseItem): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: CommentResponseItem, newItem: CommentResponseItem): Boolean {
        return oldItem == newItem
    }
}